Add string check for enums

Naragato opened the pull request:
#459

Naragato commented:
Actually slight error in my PR will correct now :)

bgrainger commented:
Please add a test that verifies the new behaviour.

Naragato commented:
Done. : >

bgrainger commented:
Sorry, I wasn't clear: the test needs to be in the SideBySide project to verify that the behaviour is the same in MySqlConnector and Connector/NET (since this is trying to clone existing Connector behaviour).

Please also revert the changes to the key files.

Naragato commented:
Reverted the certs (not sure what actually changed them :/ I left the original test in place as it makes sense to be there. However I added an additional test into the sidebyside project.

bgrainger commented on src/MySqlConnector/MySql.Data.MySqlClient/MySqlParameter.cs:
```diff
@@ -303,6 +303,10 @@ internal void AppendSqlString(BinaryWriter writer, StatementPreparerOptions opti
 			{
 				writer.WriteUtf8("{0}".FormatInvariant((ulong) Value));
 			}
+			else if (MySqlDbType == MySqlDbType.String || MySqlDbType == MySqlDbType.VarChar && HasSetDbType && Value is Enum)
```
Precedence requires parentheses around `(MySqlDbType == MySqlDbType.String || MySqlDbType == MySqlDbType.VarChar)`.

bgrainger commented:
Thanks for the bug report and PR!
